Original patch from: ../crosstool-NG/ct-ng.trunk/patches/binutils/2.18.50.0.4/130-i386-no-divide-comment.patch

-= BEGIN original header =-
-= END original header =-

diff -durN binutils-2.18.50.0.4.orig/gas/config/tc-i386.c binutils-2.18.50.0.4/gas/config/tc-i386.c
--- binutils-2.18.50.0.4.orig/gas/config/tc-i386.c	2008-02-08 17:44:10.000000000 +0100
+++ binutils-2.18.50.0.4/gas/config/tc-i386.c	2009-03-07 12:17:31.000000000 +0100
@@ -300,24 +300,10 @@
 #endif
 	;
 
-#if (defined (TE_I386AIX)				\
-     || ((defined (OBJ_ELF) || defined (OBJ_MAYBE_ELF))	\
-	 && !defined (TE_GNU)				\
-	 && !defined (TE_LINUX)				\
- 	 && !defined (TE_NETWARE)			\
-	 && !defined (TE_FreeBSD)			\
-	 && !defined (TE_NetBSD)))
 /* This array holds the chars that always start a comment.  If the
-   pre-processor is disabled, these aren't very useful.  The option
-   --divide will remove '/' from this list.  */
-const char *i386_comment_chars = "#/";
-#define SVR4_COMMENT_CHARS 1
-#define PREFIX_SEPARATOR '\\'
-
-#else
+   pre-processor is disabled, these aren't very useful.  */
 const char *i386_comment_chars = "#";
 #define PREFIX_SEPARATOR '/'
-#endif
 
 /* This array holds the chars that only start a comment at the beginning of
    a line.  If the line seems to have the form '# 123 filename'
@@ -7243,20 +7229,6 @@
       break;
 
     case OPTION_DIVIDE:
-#ifdef SVR4_COMMENT_CHARS
-      {
-	char *n, *t;
-	const char *s;
-
-	n = (char *) xmalloc (strlen (i386_comment_chars) + 1);
-	t = n;
-	for (s = i386_comment_chars; *s != '\0'; s++)
-	  if (*s != '/')
-	    *t++ = *s;
-	*t = '\0';
-	i386_comment_chars = n;
-      }
-#endif
       break;
 
     case OPTION_MARCH:
@@ -7392,13 +7364,8 @@
   fprintf (stream, _("\
   --32/--64               generate 32bit/64bit code\n"));
 #endif
-#ifdef SVR4_COMMENT_CHARS
-  fprintf (stream, _("\
-  --divide                do not treat `/' as a comment character\n"));
-#else
   fprintf (stream, _("\
   --divide                ignored\n"));
-#endif
   fprintf (stream, _("\
   -march=CPU[,+EXTENSION...]\n\
                           generate code for CPU and EXTENSION, CPU is one of:\n\
